src/components/Sidebar/Sidebar.jsx

```diff
@@ -10,90 +10,220 @@ import {
   Box,
   Text,
   HStack,
+  useDisclosure,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalCloseButton,
+  ModalBody,
+  Spacer,
+  Icon,
 } from '@chakra-ui/react';
 import 'boxicons';
 import fphLogo from './fph-logo.png.png';
+import { useAuth } from '../../contexts/AuthContext';
+import { useBackend } from '../../contexts/BackendContext';
+import { useEffect, useState } from 'react';
+import { BiBuildingHouse, BiPhone, BiEnvelope } from 'react-icons/bi';
+
+function Sidebar() {
+  const { backend } = useBackend();
+  const [isAdminUser, setIsAdminUser] = useState(false);
+  const { isOpen, onOpen, onClose } = useDisclosure(); // State management for modal
+
+  useEffect(() => {
+    const checkIsAdmin = async () => {
+      if (await isAdmin()) {
+        setIsAdminUser(true);
+      }
+    };
+
+    checkIsAdmin();
+  });
+
+  useEffect(() => {
+    getBusinessName();
+  },
+  [backend]);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has a missing dependency: 'getBusinessName'. Either include it or remove the dependency array.